### PR TITLE
[SPARK-7394][SQL] Add Pandas style cast (astype)

### PR DESCRIPTION
**Pull request by kaka1992**

### CODE REVIEW

File: python/pyspark/sql/column.py [hunk 1]
```diff
@@ -294,6 +294,17 @@ def cast(self, dataType):
         return Column(jc)
 
     @ignore_unicode_prefix
+    def astype(self, dataType):
```
Comment (rxin) [inline python/pyspark/sql/column.py]: you can just do
```python
astype = cast
```

and I think everything will work.

cc @davies

Comment (rxin) [inline python/pyspark/sql/column.py]: (see the groupby function at the end of DataFrame)

**Comment by davies** [comment]
Jenkins, OK to test

**Comment by davies** [comment]
LGTM

**Pull request closed without merging**